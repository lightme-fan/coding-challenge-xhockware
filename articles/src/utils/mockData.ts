export const article = {
    id: "123",
    source: {
    id: "cnbc",
    name: "CNBC"
  },
    author: "Amelia Lucas",
    title: "Coca-Cola stock jumps as strong sales of Coke Zero Sugar continue to drive revenue growth - CNBC",
    description: "Coca-Cola on Friday reported quarterly earnings that met analysts' expectations.",
    url: "https://www.cnbc.com/2019/10/18/coca-cola-ko-earnings-q3-2019.html",
    urlToImage: "https://image.cnbcfm.com/api/v1/image/105375236-1533317591071rtx6drjw.jpg?v=1571332846",
    publishedAt: "2019-10-18T10:42:46Z",
    content: "Coca-Cola on Friday reported quarterly revenue.",
    viewText: "Read"
}