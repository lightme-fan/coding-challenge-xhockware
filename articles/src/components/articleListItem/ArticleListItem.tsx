import React, { FC } from 'react';
import styles from "./styles.module.css";
import { Article } from '../../utils/types';

const {
  item_container,
  image_wrapper,
  image_style,
  source,
  writer,
  heading,
  publishedAt,
  author
} = styles;

interface ListItemType {
  article: Article;
  handleClickOnArticle: (value: any) => void;
}

const ArticleListItem : FC<ListItemType>= ({ article, handleClickOnArticle }) => {
  return (
    <li className={item_container} onClick={() => handleClickOnArticle(article.id)} data-testid="article">
      <div className={image_wrapper}>
        <img className={image_style} src={article.urlToImage} alt="Newspaper image" />
      </div>
      <div className={source}>
        <h2 className={heading}>{article.title}</h2>
        <div className={writer}>
          <div className={publishedAt}>Published in: {article.publishedAt}</div>
          <div className={author}>{article.source.name}</div>
          <div data-testid="viewText"><b><i>{article.viewText} article</i></b></div>
        </div>
      </div>
    </li>
  )
}

export default ArticleListItem