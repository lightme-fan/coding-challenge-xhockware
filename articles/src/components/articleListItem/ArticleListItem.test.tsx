import userEvent from '@testing-library/user-event'
import '@testing-library/jest-dom'
import { render } from '@testing-library/react'
import ArticleListItem from './ArticleListItem'
import { article } from '../../utils/mockData'

describe("Tests an article ", () => {
  const handleClickOnArticle = jest.fn();

  test('tests list items', async () => {
    const component = render(
    <ArticleListItem
      key={article.id}
      article={article}
      handleClickOnArticle={() => handleClickOnArticle(article.id)}
    />
  )
    const { getByTestId } = component;
    const listItem = getByTestId("article");
    const viewText = getByTestId("viewText");
    expect(listItem).toContainHTML("div");

    await userEvent.click(listItem);
    await expect(handleClickOnArticle).toHaveBeenCalled();
    await expect(viewText).toHaveTextContent(`${article.viewText} article`);

    expect(component).toMatchSnapshot();
  })
})