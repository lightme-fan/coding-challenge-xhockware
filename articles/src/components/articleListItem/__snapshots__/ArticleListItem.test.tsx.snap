// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests an article  tests if article details is not null 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <li
        class="item_container"
        data-testid="article"
      >
        <div
          class="image_wrapper"
        >
          <img
            alt="Newspaper image"
            class="image_style"
            src="https://image.cnbcfm.com/api/v1/image/105375236-1533317591071rtx6drjw.jpg?v=1571332846"
          />
        </div>
        <div
          class="source"
        >
          <h2
            class="heading"
          >
            Coca-Cola stock jumps as strong sales of Coke Zero Sugar continue to drive revenue growth - CNBC
          </h2>
          <div
            class="writer"
          >
            <div
              class="publishedAt"
            >
              Published in: 
              2019-10-18T10:42:46Z
            </div>
            <div
              class="author"
            >
              CNBC
            </div>
            <div>
              <b>
                <i>
                  Unread
                   article
                </i>
              </b>
            </div>
          </div>
        </div>
      </li>
    </div>
  </body>,
  "container": <div>
    <li
      class="item_container"
      data-testid="article"
    >
      <div
        class="image_wrapper"
      >
        <img
          alt="Newspaper image"
          class="image_style"
          src="https://image.cnbcfm.com/api/v1/image/105375236-1533317591071rtx6drjw.jpg?v=1571332846"
        />
      </div>
      <div
        class="source"
      >
        <h2
          class="heading"
        >
          Coca-Cola stock jumps as strong sales of Coke Zero Sugar continue to drive revenue growth - CNBC
        </h2>
        <div
          class="writer"
        >
          <div
            class="publishedAt"
          >
            Published in: 
            2019-10-18T10:42:46Z
          </div>
          <div
            class="author"
          >
            CNBC
          </div>
          <div>
            <b>
              <i>
                Unread
                 article
              </i>
            </b>
          </div>
        </div>
      </div>
    </li>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Tests an article  tests list items 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <li
        class="item_container"
        data-testid="article"
      >
        <div
          class="image_wrapper"
        >
          <img
            alt="Newspaper image"
            class="image_style"
            src="https://image.cnbcfm.com/api/v1/image/105375236-1533317591071rtx6drjw.jpg?v=1571332846"
          />
        </div>
        <div
          class="source"
        >
          <h2
            class="heading"
          >
            Coca-Cola stock jumps as strong sales of Coke Zero Sugar continue to drive revenue growth - CNBC
          </h2>
          <div
            class="writer"
          >
            <div
              class="publishedAt"
            >
              Published in: 
              2019-10-18T10:42:46Z
            </div>
            <div
              class="author"
            >
              CNBC
            </div>
            <div
              data-testid="viewText"
            >
              <b>
                <i>
                  Read
                   article
                </i>
              </b>
            </div>
          </div>
        </div>
      </li>
    </div>
  </body>,
  "container": <div>
    <li
      class="item_container"
      data-testid="article"
    >
      <div
        class="image_wrapper"
      >
        <img
          alt="Newspaper image"
          class="image_style"
          src="https://image.cnbcfm.com/api/v1/image/105375236-1533317591071rtx6drjw.jpg?v=1571332846"
        />
      </div>
      <div
        class="source"
      >
        <h2
          class="heading"
        >
          Coca-Cola stock jumps as strong sales of Coke Zero Sugar continue to drive revenue growth - CNBC
        </h2>
        <div
          class="writer"
        >
          <div
            class="publishedAt"
          >
            Published in: 
            2019-10-18T10:42:46Z
          </div>
          <div
            class="author"
          >
            CNBC
          </div>
          <div
            data-testid="viewText"
          >
            <b>
              <i>
                Read
                 article
              </i>
            </b>
          </div>
        </div>
      </div>
    </li>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
